model_compression_toolkit/core/common/graph/memory_graph/memory_graph.py

```diff
@@ -71,6 +82,7 @@ def __init__(self, model_graph: Graph):
         inputs_tensors_memory = [sum([t.total_size for t in self.operation_node_children(n)])
                                  for n in nodes if n in model_graph.get_inputs()]
 
+        # TODO maxcut: why both inputs and outputs of each nodes, while the A* solves for node outputs only???
```

Comment:
We need to deeply get into the algorithm to understand this.
If it is not giving you trouble then leave it for now.
Or, create some unittest that you for sure know what the result should be and see if changing this code according to what you think it should be changes the result.